File: src/data/players.ts
export const PLAYER_COLORS = [
  '#E74C3C',
  '#3498DB',
  '#2ECC71',
  '#F39C12',
  '#9B59B6',
  '#E67E22',
]

const HEX_COLOR_RE = /^#([0-9a-f]{3}|[0-9a-f]{6}|[0-9a-f]{8})$/i;

export function isValidColor(value: unknown): value is string {
  return typeof value === 'string' && HEX_COLOR_RE.test(value);
}

export function normalizeColor(value: string): string {
  const hex = value.toLowerCase();
  if (hex.length === 4) {
    return '#' + hex[1] + hex[1] + hex[2] + hex[2] + hex[3] + hex[3];
  }
  if (hex.length === 9) {
    return '#' + hex[1] + hex[2] + hex[3] + hex[4] + hex[5] + hex[6] + hex[7] + hex[8];
  }
  return hex;
}

export const MAX_PLAYERS = 6

export const PLAYER_OFFSETS: Record<number, { dx: number; dy: number }> = {
  0: { dx: -8, dy: -8 },
  1: { dx: 8, dy: -8 },
  2: { dx: -8, dy: 8 },
  3: { dx: 8, dy: 8 },
  4: { dx: 0, dy: -8 },
  5: { dx: 0, dy: 8 },
}

File: backend/internal/data/players.go
package data

import (
	"regexp"
	"strings"
)

var PLAYER_COLORS = []string{
	"#E74C3C",
	"#3498DB",
	"#2ECC71",
	"#F39C12",
	"#9B59B6",
	"#795548",
}

const MAX_PLAYERS = 6

type PlayerOffset struct {
	Dx int `json:"dx"`
	Dy int `json:"dy"`
}

var PLAYER_OFFSETS = map[int]PlayerOffset{
	0: {Dx: -8, Dy: -8},
	1: {Dx: 8, Dy: -8},
	2: {Dx: -8, Dy: 8},
	3: {Dx: 8, Dy: 8},
	4: {Dx: 0, Dy: -8},
	5: {Dx: 0, Dy: 8},
}

var hexColorRe = regexp.MustCompile(`^#([0-9a-fA-F]{3}|[0-9a-fA-F]{6}|[0-9a-fA-F]{8})$`)

func IsValidColor(value string) bool {
	return hexColorRe.MatchString(value)
}

func NormalizeColor(value string) string {
	hex := strings.ToLower(value)
	if len(hex) == 4 {
		return "#" + string(hex[1]) + string(hex[1]) + string(hex[2]) + string(hex[2]) + string(hex[3]) + string(hex[3])
	}
	return hex
}
